### DISCUSSION TITLE
JetStream NATSNoRespondersException

### DISCUSSION THREAD

**Discussion by Walluce** [nats-io/nats.net.v1#628] [Q&A] [answered]
Question about NATSNoRespondersException behaviour. 

We are considering using NATS and JetStream for inter component communication. 

I had assumed that if I write a message to a stream, it will get saved on the stream for consumer to pick up later (like Kafka). Instead I am getting an exception. 

What I am concerned about is the use-case where all of my consumers are stopped / fall over. I would not like to lose messages because the consumer can't produce any more.

What am I doing wrong?

**Comment by scottf** [comment]
@Walluce No Responders means the server cannot find any stream with the subject you are publishing to. What is your stream configuration and what subject are you publishing to?

**Reply by Walluce** [reply]
Hi, @scottf, 
The stream name is "users"
The subjects specified when I created the stream was "users.*"
The subject of the message I am publishing is "users.logoff.[random integer]"

I am trying to set up Deterministic Token Partitioning 

This is my config file:

```
mappings: {
    foo : bar
    "users.logoff.*" : "users.logoff.{{wildcard(1)}}.{{partition(10,1)}"
    "neworders.*" : "neworders.{{wildcard(1)}}.{{partition(3,1)}}"
}
```

**Comment by scottf** [comment, marked as answer]
I see the problem. The `*`, only represents one segment. So `users.*` only allows for `users.a` not `users.a.b`
If you specify `users.>` as your stream subject, you will be fine. You could I think specify `users.*.*` 

Here is the documentation for subjects https://docs.nats.io/nats-concepts/subjects#subject-hierarchies

**Comment by Walluce** [comment]
Thank you, that solved my problems.
